### PR TITLE
Tests and handling of comments/spaces

### PR DESCRIPTION
**Pull request by x-way**
Started to add some additional tests based on https://git-scm.com/docs/gitignore
Then saw that the handling of comments and trailing whitespaces is not exactly following the spec yet, thus added some code to handle these.

### CODE REVIEW

File: gitignore_parser.py [hunk 1]
```diff
@@ -87,6 +87,21 @@ def rule_from_pattern(pattern, base_path=None, source=None):
 		pattern = pattern[1:]
 	if pattern[-1] == '/':
 		pattern = pattern[:-1]
+        # patterns with leading hashes are escaped with a backslash in front, unescape it
```
[the remark this answers is not in the document]
Comment (x-way) [reply gitignore_parser.py]: Thanks for spotting this, fixed.

File: gitignore_parser.py [hunk 2]
```diff
@@ -87,6 +87,21 @@ def rule_from_pattern(pattern, base_path=None, source=None):
...
+	if pattern[0] == '\\' and pattern[1] == '#':
+		pattern = pattern[1:]
+        # trailing spaces are ignored unless they are escaped with a backslash
```
[the remark this answers is not in the document]
Comment (x-way) [reply gitignore_parser.py]: Thanks for spotting this, fixed.

**Comment by mherrmann** [comment]
Is this already ready to be merged @x-way?

**Comment by x-way** [comment]
Yes, this is ready to be merged now.

**Comment by inverse** [comment]
@x-way if you rebase/merge master you'll get CI running your tests ;)

**Comment by bitranox** [comment]
@mherrmann You were a little bit quicker then me - I also merged that into `igittigitt` v1.0.4

I doubt that handling trailing spaces makes some sense, because there can not be files or directories with trailing spaces on the filesystem ? 
So such trailing spaces should not occur right ? 
Or do I miss something ?

**Comment by ahernsean** [comment]
@bitranox Files and directories can easily have trailing spaces on the file system (I just tried it):
```
% touch "trailingspace "
% ls
trailingspace
% rm "trailingspace"
rm: cannot remove ‘trailingspace’: No such file or directory
% rm "trailingspace "
%
```

**Pull request merged**